app/1.0.x/upgrading.md

```diff
@@ -0,0 +1,285 @@
+    for more details.
+
+## 2. Deprecation Notices
+
+There are no deprecation notices in this release.
+
+## 3. Suggested Upgrade Path
+
+### Preliminary Checks
+
+If your cluster is running a version lower than 0.14, you need to
+upgrade to 0.14.1 first instead. Upgrading from a pre-0.14 cluster
+straight to Kong 1.0 is **not** supported.
+
+If you still use the deprecated API entity to configure your endpoints and
+upstream services (via `/apis`) instead of using Routes for endpoints (via
+`/routes`) and Services for upstream services (via `/services`), now is the
+time to do so. Kong 1.0 will refuse to run migrations if you have any entity
+configured using `/apis` in your datastore. Create equivalent Routes and
+Services and delete your APIs. (Note that Kong does not do this automatically
+because the naive option of creating a Route and Service pair for each API
+would miss the point of the improvements brought by Routes and Services;
+the ideal mapping of Routes and Services depends on your microservice
+architecture.)
+
+If you use additional plugins other than the ones bundled with Kong,
+make sure they are compatible with Kong 1.0 prior to upgrading.
+See the section above on Plugins for information on plugin compatibility.
+
+### Migration Steps from 0.14
+
+Kong 1.0 introduces a new, improved migrations framework.
+It supports a no-downtime, Blue/Green migration model for upgrading
+from 0.14.x. The full migration is now split into two steps,
+which are performed via commands `kong migrations up` and
+`kong migrations finish`.
+
+For a no-downtime migration from a 0.14 cluster to a 1.0 cluster,
+we recommend the following sequence of steps:
+
+1. Download 1.0, and configure it to point to the same datastore
+   as your 0.14 cluster. Run `kong migrations up`.
+2. Both 0.14 and 1.0 nodes can now run simultaneously on the same
+   datastore. Start provisioning 1.0 nodes, but do not use their
+   Admin API yet. Prefer making Admin API requests to your 0.14 nodes
+   instead.
+3. Gradually divert traffic away from your 0.14 nodes, and into
+   your 1.0 cluster. Monitor your traffic to make sure everything
+   is going smoothly.
+4. When your traffic is fully migrated to the 1.0 cluster,
+   decommission your 0.14 nodes.
+5. From your 1.0 cluster, run: `kong migrations finish`.
+   From this point on, it will not be possible to start 0.14
+   nodes pointing to the same datastore anymore. Only run
+   this command when you are confident that your migration
+   was successful. From now on, you can safely make Admin API
+   requests to your 1.0 nodes.
+
+### Upgrade Path from 1.0 Release Candidates
+
+The process is the same as for upgrading for 0.14 listed above, but on step 1 you should run `kong migrations up --force` instead.
+
+### Upgrade Path for patch releases
+
+There are currently no point releases of kong 1.0.0 (there is no 1.0.1, 1.0.2, etc.) Once those are available,
+the update process will be the following:
```

Comment:
Kong is uppercase. Also, this sentence is bound to be incorrect once 1.0.1 is out; better to write it in a future-proof way:

```
There are no migrations in upgrades between current or future patch releases of the same minor release of Kong (e.g. 1.0.0 to 1.0.1, 1.0.1 to 1.0.4, etc.). Therefore, the upgrade process is simpler.
```